src/shared.cpp

```diff
@@ -1431,6 +1431,34 @@ QSet<QString> codesignBundle(const QString &identity,
 
     bool getAbsolutePath = true;
     QStringList foundPluginBinaries = findAppBundleFiles(appBundlePath + "/Contents/PlugIns/", getAbsolutePath);
+
+    // Check if we have any .appex (extension) bundles in the PlugIns directory
+    QStringList::size_type appexIndex;
+    QString pluginDirBasePath = appBundlePath + "/Contents/PlugIns/";
+    while ((appexIndex = foundPluginBinaries.indexOf(QRegularExpression("^" + pluginDirBasePath + "\\w+\\.appex\\/.*"))) != -1)
```

Comment:
Instead of iterating over all files in the binaries list, we could add a second list using a new function similar to the [Framework list function](https://github.com/GDATASoftwareAG/macdeployqt/blob/6f324209679de5ea8c645c5b8aa32d0a8eaef8f6/src/shared.cpp#L439-L451), which should then only contain all .appex dirs. This will save us from doing all that RegEx magic, we don't have to care about the iterator (can use a range-based loop) and still use `removeIf` as done below to delete the files from the `foundPluginBinaries` list. IMO this would make the code way less messy.

Reply:
You're right, it might break if there is an extension inside an extension. Can that really happen? I don't know, but it would break if I find the outer bundle before I find the inner bundle.

Reply:
@kblaschke Changed, should work now.